pinot-spi/src/main/java/org/apache/pinot/spi/utils/builder/TableConfigBuilder.java

```diff
@@ -256,6 +258,16 @@ public TableConfigBuilder setInvertedIndexColumns(List<String> invertedIndexColu
     return this;
   }
 
+  public TableConfigBuilder setOptimizeDictionaryForMetrics(boolean optimizeDictionaryForMetrics) {
+    _optimizeDictionaryForMetrics = optimizeDictionaryForMetrics;
+    return this;
+  }
+
+  public TableConfigBuilder setNoDictionarySizeRatioThreshold(double noDictionarySizeRatioThreshold) {
```

Comment:
This naming is pretty confusing. 

"sizeThresholdForNoDictionary" or "ratioForNoDictionary" ?

Reply:
Im following what we introduced in the tableIndexConfig, to keep it consistent :).

From this doc - https://docs.pinot.apache.org/configuration-reference/table#table-index-config
"If optimizeDictionaryForMetrics enabled, dictionary is not created for the metric columns for which noDictionaryIndexSize/ indexWithDictionarySize is less than the noDictionarySizeRatioThreshold
Default: 0.85"

Reply:
Im following what we introduced in the index table config :)
https://docs.pinot.apache.org/configuration-reference/table#table-index-config

> does this test include data? can we add a test with some data and verify that the threshold is honored or not?
> 
> Otherwise, lgtm!

The test creates tableConfig with some values in the 2 new fields, serializes & de-serializes the tableConfig and verifies if the values match with what was initialized.

Reply:
>Im following what we introduced in the tableIndexConfig, to keep it consistent :).

Yeah. But it is not clear. At first glance, `optimizeDictionaryForMetrics` made me think " for what metrics are we optimizing? " :)) If we want to be this verbose, then it should have been `optimizeDictionaryForMetricColumns` :)) 

Would really appreciate an improvement over `noDictionarySizeRatioThreshold`. Welcome others to chime in as well.


> The test creates tableConfig with some values in the 2 new fields, serializes & de-serializes the tableConfig and verifies if the values match with what was initialized.

Right. That is only testing the config values. do we not already have tests that loads a metric column with data and ensures the indexing config actually enforces the optimization? I would assume such a test would have been added when this optimization was introduced.

Reply:
Yes thats part of this [PR](https://github.com/apache/pinot/pull/8398), to read from the IndexConfig.